build/questbook.py

```diff
@@ -103,54 +103,43 @@ def nest(location: dict) -> dict:
     return location
 
 
-def i18n(output: dict, book: dict, location: str, place: str, prefix: str):
+def i18n(output: dict, id: int, entry: dict, place: str, prefix: str):
     """converts questbook title/desc into lang file"""
 
     start = "%s.quest" % prefix
 
     alreadyKnownKeys = []
-    for entry in dict(book[location]):
-        if ("questID:3" in book[location][entry]):
-            id = book[location][entry]["questID:3"]
-        elif ("lineID:3" in book[location][entry]):
-            id = book[location][entry]["lineID:3"]
-        else:
-            print("Could not find questid or lineid for location %s %s" % (location, entry))
-            continue
-
-        key = "%s.%s.%s" % (start, place, id)
-        title = key + ".title"
-        desc = key + ".desc"
-
-        if (book[location][entry]["properties:10"]["betterquesting:10"]["name:8"].startswith(start)):
-            alreadyKnownKeys.append(title)
-        else:
-            output[title] = convertToLang(book[location][entry]["properties:10"]["betterquesting:10"]["name:8"])
-            book[location][entry]["properties:10"]["betterquesting:10"]["name:8"] = title.rstrip()
-
-        if (book[location][entry]["properties:10"]["betterquesting:10"]["desc:8"].startswith(start)):
-            alreadyKnownKeys.append(desc)
-        else:
-            output[desc] = convertToLang(book[location][entry]["properties:10"]["betterquesting:10"]["desc:8"])
-            book[location][entry]["properties:10"]["betterquesting:10"]["desc:8"] = desc.rstrip()
+    
+    key = "%s.%s.%s" % (start, place, id)
+    title = key + ".title"
+    desc = key + ".desc"
+
+    if (entry["properties:10"]["betterquesting:10"]["name:8"].startswith(start)):
+        alreadyKnownKeys.append(title)
+    else:
+        output[title] = convertToLang(entry["properties:10"]["betterquesting:10"]["name:8"])
+        entry["properties:10"]["betterquesting:10"]["name:8"] = title.rstrip()
+        
+    if (entry["properties:10"]["betterquesting:10"]["desc:8"].startswith(start)):
+        alreadyKnownKeys.append(desc)
+    else:
+        output[desc] = convertToLang(entry["properties:10"]["betterquesting:10"]["desc:8"])
+        entry["properties:10"]["betterquesting:10"]["desc:8"] = desc.rstrip()
 
     if (len(alreadyKnownKeys) > 0):
         print("Already knew %s keys " % (len(alreadyKnownKeys)))
```

Comment:
This particular log output is spammed across every single file; could this be changed as to not log anything, or just to log this once at the end of the file-scanning process?

Reply:
done